app/components/AndroidListView.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import {
+  AppRegistry,
+  ListView,
+  StyleSheet,
+  Text,
+  View,
+  TouchableHighlight
+} from 'react-native';
+
+import mockData from '../mockData.json' // TEMP
+import { ChallengeDetailRoute } from '../routes/defaultRoutes'
+
+export default class AndroidListView extends Component {
+  constructor(props) {
+    super(props);
+    this.goToChallengeDetailScene = this.goToChallengeDetailScene.bind(this);
+    this.dataSource = new ListView.DataSource({
+      rowHasChanged: (r1, r2) => r1 !== r2
+    })
+    this.state = {
+      mockData: mockData.challenges
+    };
+  }
+
+  goToChallengeDetailScene() {
+    var selectedChallenge = mockData.challenges[0]; // TEMP CHALLENGE MOCK
+
+    // build route
+    var route = ChallengeDetailRoute;
```

Comment:
@jwt1143 thanks for the explanation, I hadn't considered all those aspects. I'll have to update mine to follow this pattern.